core/domain/suggestion_domain.py

```diff
@@ -72,3 +66,196 @@ def to_dict(self):
+                are invalid.
+        """
+        if (
+                self.suggestion_type not in
+                suggestion_models.SUGGESTION_TYPE_CHOICES):
+            raise utils.ValidationError(
+                'Expected suggestion_type to be among allowed choices, '
+                'recieved %s' % self.suggestion_type)
+
+        if self.target_type not in suggestion_models.TARGET_TYPE_CHOICES:
+            raise utils.ValidationError(
+                'Expected target_type to be among allowed choices, '
+                'recieved %s' % self.target_type)
+
+        if not isinstance(self.target_id, basestring):
+            raise utils.ValidationError(
+                'Expected target_id to be a string, recieved %s' % type(
+                    self.target_id))
+
+        if not isinstance(self.target_version_at_submission, int):
+            raise utils.ValidationError(
+                'Expected target_version_at_submission to be an int, '
+                'recieved %s' % type(self.target_version_at_submission))
+
+        if self.status not in suggestion_models.STATUS_CHOICES:
+            raise utils.ValidationError(
+                'Expected status to be among allowed choices, '
+                'recieved %s' % self.status)
+
+        if not isinstance(self.author_id, basestring):
+            raise utils.ValidationError(
+                'Expected author_id to be a string, recieved %s' % type(
+                    self.author_id))
+
+        if not isinstance(self.assigned_reviewer_id, basestring):
+            raise utils.ValidationError(
+                'Expected assigned_reviewer_id to be a string,'
+                ' recieved %s' % type(self.assigned_reviewer_id))
+
+        if not isinstance(self.final_reviewer_id, basestring):
+            raise utils.ValidationError(
+                'Expected final_reviewer_id to be a string, recieved %s' % type(
+                    self.final_reviewer_id))
+
+        if not isinstance(self.change_cmd, dict):
+            raise utils.ValidationError(
+                'Expected change_cmd to be a dict, recieved %s' % type(
+                    self.change_cmd))
+
+        if not isinstance(self.score_category, basestring):
+            raise utils.ValidationError(
+                'Expected score_category to be a string, recieved %s' % type(
+                    self.score_category))
+
+        if (
+                not suggestion_models.SCORE_CATEGORY_DELIMITER in
+                self.score_category):
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, recieved %s' %
+                suggestion_models.SCORE_CATEGORY_DELIMITER, type(
+                    self.score_category))
+
+        if (
+                self.score_category.split(
+                    suggestion_models.SCORE_CATEGORY_DELIMITER)[0] not in
```

Comment:
You also need to check that there are exactly two parts and validate the second part as well (perhaps in subclasses)?

Reply:
Done. Added validations for both parts in the sub class.